samples/aws/external/cluster-template-external-cloud-provider.yaml

```diff
@@ -30,102 +25,150 @@ apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
 kind: AWSCluster
 metadata:
   name: ${CLUSTER_NAME}
-  namespace: ${CABPR_NAMESPACE}
+  namespace: ${CLUSTER_NAMESPACE}
 spec:
+  bastion:
+    enabled: true
   network:
     vpc:
       availabilityZoneUsageLimit: 1
-  bastion:
-    enabled: true
+    additionalControlPlaneIngressRules:
+      - description: "RKE2 Control Plane additional group"
+        protocol: "tcp"
+        fromPort: 9345
+        toPort: 9345
+        sourceSecurityGroupRoles:
+          - node
+          - controlplane
+  controlPlaneLoadBalancer:
+      loadBalancerType: nlb
+      additionalListeners:
+        - port: 9345
+          protocol: "TCP"
+      ingressRules:
+      - description: RKE2 Control Plane
+        fromPort: 9345
+        protocol: tcp
+        sourceSecurityGroupRoles:
+        - node
+        - controlplane
+        toPort: 9345
+      - description: Kubernetes API
+        fromPort: 6443
+        protocol: tcp
+        toPort: 6443
+        cidrBlocks: 
+        - "0.0.0.0/0"
   region: ${AWS_REGION}
   sshKeyName: ${AWS_SSH_KEY_NAME}
+
 ---
 apiVersion: controlplane.cluster.x-k8s.io/v1beta1
 kind: RKE2ControlPlane
 metadata:
   name: ${CLUSTER_NAME}-control-plane
-  namespace: ${CABPR_NAMESPACE}
-spec: 
+  namespace: ${CLUSTER_NAMESPACE}
+spec:
+  version: ${RKE2_VERSION}
   preRKE2Commands:
   - sudo hostnamectl set-hostname $(curl -s http://169.254.169.254/1.0/meta-data/hostname)
-  replicas: ${CABPR_CP_REPLICAS}
-  version: ${KUBERNETES_VERSION}+rke2r1
-  serverConfig:
-    cni: calico
-    cloudProviderName: external
+  agentConfig:
+    airGapped: true
```

Comment:
One question - are we sure that AWS will always have to be airGapped? This forces us to release AWS IAM images with each released version, if I understand it correctly.

Reply:
I generally like the idea of using elemental-toolkit for something. I agree it makes sense to explore process in more details and see how their image building solution may aid us in our scenarios.